File: RootTools2/src/main/java/com/rarnu/tools/neo/fragment/BuildPropEditFragment.java
package com.rarnu.tools.neo.fragment;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.Menu;
import android.view.View;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.TextView;
import com.rarnu.tools.neo.R;
import com.rarnu.tools.neo.base.BaseDialogFragment;
import com.rarnu.tools.neo.data.BuildPropInfo;

public class BuildPropEditFragment extends BaseDialogFragment implements View.OnClickListener {

    private TextView tvHead = null;
    private ImageView btnCancel = null;
    private ImageView btnSave = null;
    private EditText etValue = null;

    @Override
    public int getBarTitle() {
        return 0;
    }

    @Override
    public String getCustomTitle() {
        return null;
    }

    @Override
    public void initComponents() {
        tvHead = (TextView) innerView.findViewById(R.id.tvHead);
        btnSave = (ImageView) innerView.findViewById(R.id.btnSave);
        btnCancel = (ImageView) innerView.findViewById(R.id.btnCancel);
        etValue = (EditText) innerView.findViewById(R.id.etValue);
    }

    @Override
    public void initEvents() {
        btnCancel.setOnClickListener(this);
        btnSave.setOnClickListener(this);
    }

    @Override
    public void initLogic() {
        BuildPropInfo item = (BuildPropInfo) getActivity().getIntent().getSerializableExtra("item");
        tvHead.setText(item.buildName);
        etValue.setText(item.buildValue);
    }

    @Override
    public int getFragmentLayoutResId() {
        return R.layout.fragment_fakedev_edit;
    }

    @Override
    public String getMainActivityName() {
        return null;
    }

    @Override
    public void initMenu(Menu menu) {

    }

    @Override
    public void onGetNewArguments(Bundle bn) {

    }

    @Override
    public Bundle getFragmentState() {
        return null;
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.btnSave:
                BuildPropInfo info = new BuildPropInfo();
                info.buildName = tvHead.getText().toString();
                info.buildValue = etValue.getText().toString();
                Intent inRet = new Intent();
                inRet.putExtra("item", info);
                inRet.putExtra("position", getActivity().getIntent().getIntExtra("position", -1));
                getActivity().setResult(Activity.RESULT_OK, inRet);
                getActivity().finish();
                break;
            case R.id.btnCancel:
                getActivity().finish();
                break;
        }
    }
}

File: RootTools2/src/main/kotlin/com/rarnu/tools/neo/fragment/BuildPropEditFragment.kt
package com.rarnu.tools.neo.fragment

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.view.Menu
import android.view.View
import com.rarnu.base.app.BaseDialogFragment
import com.rarnu.tools.neo.R
import com.rarnu.tools.neo.data.BuildPropInfo
import kotlinx.android.synthetic.main.fragment_fakedev_edit.view.*

class BuildPropEditFragment : BaseDialogFragment(), View.OnClickListener {

    override fun getBarTitle(): Int = 0

    override fun getBarTitleWithPath(): Int = 0

    override fun getCustomTitle(): String? = null

    override fun initComponents() { }

    override fun initEvents() {
        innerView.btnCancel.setOnClickListener(this)
        innerView.btnSave.setOnClickListener(this)
    }

    override fun initLogic() {
        val item = activity.intent.getSerializableExtra("item") as BuildPropInfo?
        innerView.tvHead.text = item?.buildName
        innerView.etValue.setText(item?.buildValue)
    }

    override fun getFragmentLayoutResId(): Int = R.layout.fragment_fakedev_edit

    override fun getMainActivityName(): String? = null

    override fun initMenu(menu: Menu) { }

    override fun onGetNewArguments(bn: Bundle?) { }

    override fun getFragmentState(): Bundle? = null

    override fun onClick(v: View) {
        when (v.id) {
            R.id.btnSave -> {
                val info = BuildPropInfo(innerView.tvHead.text.toString(), innerView.etValue.text.toString())
                val inRet = Intent()
                inRet.putExtra("item", info)
                inRet.putExtra("position", activity.intent.getIntExtra("position", -1))
                activity.setResult(Activity.RESULT_OK, inRet)
                activity.finish()
            }
            R.id.btnCancel -> activity.finish()
        }
    }
}
